### PR TITLE
Update the hacking guide

### PR DESCRIPTION
**Pull request by kargakis**
@fabianofranz @deads2k

### CODE REVIEW

File: docs/cli_hacking_guide.adoc [hunk 1]
```diff
@@ -104,7 +114,7 @@ When writing an usage string, make sure you cover the most important path for th
...
 * If multiple values are supported for a given argument use three dots, e.g. `KEY_1=VAL_1 ... KEY_N=VAL_N`.
-* Arguments don't have names, but we have to reference them somehow in usage. Try to be concise with the names already used by the usage of other commands. For example, these are some very recurring names: `BUILD` (meaning a build name or ID), `DEPLOYMENT` (meaning a deployment name or ID), `RESOURCE` (e.g. pod, pods, replicationcontroller, rc, deploymentconfig, dc, build, etc), `NAME`, `RESOURCE/NAME` (e.g. pod/mypodname, rc/myrcname, etc), `URL`, `TEMPLATE`, `KEY=VALUE`, `FILENAME` and so on.
+* Arguments don't have names, but we have to reference them somehow in usage. Try to be concise with the names already used by the usage of other commands. For example, these are some very recurring names: `BUILD` (meaning a build name or ID), `DEPLOYMENT` (meaning a deployment name or ID), `KIND` (e.g. pod, pods, replicationcontroller, rc, deploymentconfig, dc, build, etc), `NAME`, `KIND/NAME` (e.g. pod/mypodname, rc/myrcname, etc), `URL`, `TEMPLATE`, `KEY=VALUE`, `FILENAME` and so on.
```
Comment (deads2k) [inline docs/cli_hacking_guide.adoc]: Upstream help talks about "TYPE/NAME" and "resource types".  These aren't kinds.

Comment (kargakis) [inline docs/cli_hacking_guide.adoc]: updated to TYPE

File: pkg/cmd/cli/cmd/edit.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ saved copy to include the latest resource version.`
 func NewCmdEdit(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
 	var options EditOptions
 	cmd := &cobra.Command{
-		Use:     "edit (RESOURCE/NAME | -f FILENAME)",
+		Use:     "edit (KIND/NAME | -f FILENAME)",
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/edit.go]: type

File: pkg/cmd/cli/cmd/env.go [hunk 3]
```diff
@@ -62,7 +62,7 @@ func NewCmdEnv(fullName string, f *clientcmd.Factory, in io.Reader, out io.Write
 	var filenames kutil.StringList
 	var env kutil.StringList
 	cmd := &cobra.Command{
-		Use:     "env RESOURCE/NAME KEY_1=VAL_1 ... KEY_N=VAL_N",
+		Use:     "env KIND/NAME KEY_1=VAL_1 ... KEY_N=VAL_N",
```
Comment (deads2k) [inline pkg/cmd/cli/cmd/env.go]: `TYPE`  There's a couple others too.

Comment (kargakis) [inline pkg/cmd/cli/cmd/env.go]: ok

File: pkg/cmd/cli/cmd/edit.go (hunk 2 above)
Comment (kargakis) [inline pkg/cmd/cli/cmd/edit.go]: ok

**Comment by deads2k** [comment]
lgtm [merge]

**Comment by liggitt** [comment]
Failing test, removing from merge queue:
```
Stderr from the command:

# github.com/openshift/origin/pkg/cmd/cli/cmd
_output/local/go/src/github.com/openshift/origin/pkg/cmd/cli/cmd/deploy.go:102: too many arguments in call to options.Validate
!!! Error in hack/../hack/common.sh:131
```

**Comment by kargakis** [comment]
updated

**Comment by fabianofranz** [comment]
lgtm

**Comment by kargakis** [comment]
[merge]

**Comment by kargakis** [comment]
re[merge]

**Comment by kargakis** [comment]
[merge]?

**Pull request merged**